custom-checks/checkstyle/src/test/java/org/openhab/tools/analysis/checkstyle/test/OnlyTabIdentationInXmlFilesCheckTest.java

```diff
@@ -19,56 +19,84 @@
-
+import com.puppycrawl.tools.checkstyle.api.Configuration;
 
 /**
- * Checks whether whitespace characters are use instead of tabs in xml files 
+ * Checks whether whitespace characters are use instead of tabs in xml files
  * indentations and generates warnings in such cases.
- * 
+ *
  * @author Lyubomir Papazov - initial contribution
+ * @author Kristina Simova - Added tests for new regular expression
  */
 public class OnlyTabIdentationInXmlFilesCheckTest extends AbstractStaticCheckTest {
-    
+
     private static final String WHITESPACE_USAGE_WARNING = "There were whitespace characters used for indentation. Please use tab characters instead";
     private static final String TABIDENT_CHECK_TEST_DIRECTORY_NAME = "onlyTabIdentationInXmlFilesCheck";
     private DefaultConfiguration config;
-    
+
     @Before
     public void setUpClass() {
         config = createCheckConfig(OnlyTabIdentationInXmlFilesCheck.class);
     }
-    
+
     @Test
     public void testOneLineXmlFile() throws Exception {
         verifyXmlTabIdentation("WhiteSpacesNotUsedBeforeOpeningTags", noMessagesExpected());
     }
-    
+
     @Test
     public void testOneIncorrectLine() throws Exception {
         String[] expectedMessages = generateExpectedMessages(5, WHITESPACE_USAGE_WARNING);
         verifyXmlTabIdentation("WhiteSpaceUsedBeforeOpeningTagInOneLine", expectedMessages);
     }
-    
+
     @Test
     public void testManyIncorrectLinesOnlyShowFirstWarning() throws Exception {
         String[] expectedMessages = generateExpectedMessages(5, WHITESPACE_USAGE_WARNING);
         verifyXmlTabIdentation("WhiteSpaceUsedBeforeOpeningTagInManyLines", expectedMessages);
     }
-    
+
     @Test
     public void testManyIncorrectLinesShowAllWarnings() throws Exception {
         String[] expectedMessages = generateExpectedMessages(5, WHITESPACE_USAGE_WARNING, 6, WHITESPACE_USAGE_WARNING);
         verifyXmlTabIdentation("WhiteSpaceUsedBeforeOpeningTagInManyLines", expectedMessages, false);
     }
-    
+
+    @Test
+    public void testXmlWithEmptyLinesAndComments() throws Exception {
+        String fileName = "BasicModuleHandlerFactory";
+        verifyXmlTabIdentation(fileName, noMessagesExpected(), false);
+    }
+
+    @Test
+    public void testXmlWithMultipleLinesWithSpacesEmptyLinesAndComments() throws Exception {
+        String fileName = "ScriptEngineManager";
+        String[] expectedMessages = generateExpectedMessages(17, WHITESPACE_USAGE_WARNING, 19, WHITESPACE_USAGE_WARNING,
+                20, WHITESPACE_USAGE_WARNING, 21, WHITESPACE_USAGE_WARNING, 22, WHITESPACE_USAGE_WARNING);
+        verifyXmlTabIdentation(fileName, expectedMessages, false);
+    }
+
+    @Test
+    public void testBindingXml() throws Exception {
+        String fileName = "binding";
+        verifyXmlTabIdentation(fileName, noMessagesExpected(), false);
+    }
+
+    @Test
+    public void testThingTypesXml() throws Exception {
+        String fileName = "thing-types";
```

Comment:
Consider giving the files and tests names to express what you are actually trying to test and maybe also isolate test cases